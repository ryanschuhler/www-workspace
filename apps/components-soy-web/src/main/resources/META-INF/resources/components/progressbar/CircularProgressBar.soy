{namespace CircularProgressBar}

/**
 * Circular Progress bar
 */
{template .render}
  {@param? calculatedProgress: int}

  <svg width="24" height="24" viewport="0 0 50 50" version="1.1" xmlns="http://www.w3.org/2000/svg">
    <circle
      class="progress-indicator progress-circle"
      cx="12"
      cy="12"
      fill="none"
      r="6"
      stroke-dasharray="37.70"
      stroke-dashoffset="0"
      stroke-width="4"
    />

    <circle
      class="progress-indicator background-circle"
      cx="12"
      cy="12"
      fill="none"
      r="6"
      stroke-dasharray="37.70"
      stroke-dashoffset="-{$calculatedProgress ?: 0}"
      stroke-width="4"
    />
  </svg>
{/template}