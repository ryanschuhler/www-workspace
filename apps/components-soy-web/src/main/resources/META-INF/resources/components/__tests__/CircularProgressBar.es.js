

'use strict';

import CircularProgressBar from '../progressbar/CircularProgressBar.es';

let component;

const buttonInnerHtml = 'test-button-content';

describe('CircularProgressBar', () => {
  afterEach(() => {
    if (component) {
      component.dispose();
    }
  });

  beforeAll(() => {
		window.Liferay = {
			mdc: {
				menu: {
					MDCSimpleMenu: jest.fn()
				}
			},
			once: jest.fn()
    };
  });

  it('should render with 25%', () => {
    component = new CircularProgressBar({
      calculatedProgress: 25
    });

    expect(component).toMatchSnapshot();
  });
});