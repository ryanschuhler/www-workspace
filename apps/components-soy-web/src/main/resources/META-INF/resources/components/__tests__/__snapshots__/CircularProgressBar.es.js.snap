// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgressBar should render with 25% 1`] = `
<svg width="24" height="24" viewport="0 0 50 50" version="1.1" xmlns="http://www.w3.org/2000/svg">
  <circle class="progress-indicator progress-circle" cx="12" cy="12" fill="none" r="6" stroke-dasharray="37.70" stroke-dashoffset="0" stroke-width="4"></circle>
  <circle class="progress-indicator background-circle" cx="12" cy="12" fill="none" r="6" stroke-dasharray="37.70" stroke-dashoffset="-25" stroke-width="4"></circle>
</svg>
`;
